specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-06-01-preview/ConfidentialWatchlists.json

```diff
@@ -0,0 +1,441 @@
+      "get": {
+        "x-ms-examples": {
+          "Get a confidential watchlist.": {
+            "$ref": "./examples/confidentialWatchlists/GetConfidentialWatchlistByAlias.json"
+          }
+        },
+        "tags": [
+          "ConfidentialWatchlists"
+        ],
+        "description": "Gets a confidential watchlist, without its watchlist items.",
+        "operationId": "ConfidentialWatchlists_Get",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/WorkspaceName"
+          },
+          {
+            "$ref": "./common/WatchlistTypes.json#/parameters/WatchlistAlias"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "./common/WatchlistTypes.json#/definitions/Watchlist"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "delete": {
+        "x-ms-examples": {
+          "Delete a confidential watchlist.": {
+            "$ref": "./examples/confidentialWatchlists/DeleteConfidentialWatchlist.json"
+          }
+        },
+        "tags": [
+          "ConfidentialWatchlists"
+        ],
+        "description": "Delete a confidential watchlist.",
+        "operationId": "ConfidentialWatchlists_Delete",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/WorkspaceName"
+          },
+          {
+            "$ref": "./common/WatchlistTypes.json#/parameters/WatchlistAlias"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "headers": {
+              "Azure-AsyncOperation": {
```

Comment:
>"Azure-AsyncOperation": { [](http://example.com/codeflow?start=14&length=25)

Is DELETE async? If so, it needs to return 202 and set x-ms-long-running-operation to true.

[https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#delete-resource-asynchronously](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#delete-resource-asynchronously)

[https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-long-running-operation](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-long-running-operation)

Reply:
In previous api version ([2021-04-01](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2021-04-01/Watchlists.json#L167)), the operation was not async, and when we changed to async in 2022-01-01-preview and to return 202 () , it was flagged as breaking changes and we had to change it to return 200.

Reply:
We went to the weekly breaking changes office hours, and we did mention that the design to move from 200 to 202 has been already discussed with ARM. They did not approve the breaking changes. We had to revert it back to 200.